iac-common/src/main/java/org/sonar/iac/common/extension/visitors/SyntaxHighlightingVisitor.java

```diff
@@ -31,12 +33,10 @@ public abstract class SyntaxHighlightingVisitor extends TreeVisitor<InputFileCon
 
   private NewHighlighting newHighlighting;
 
+  private Set<TextRange> commentTextRanges = new HashSet<>();
+
   protected SyntaxHighlightingVisitor() {
-    register(Tree.class, (ctx, tree) -> {
-      if (tree instanceof HasComments) {
-        ((HasComments) tree).comments().forEach(comment -> highlight(comment, COMMENT));
-      }
-    });
+    register(Tree.class, (ctx, tree) -> highlightComments(tree));
```

Comment:
Because now each `Tree` implement `HasComments`. See: https://github.com/SonarSource/sonar-iac/pull/560/files#diff-5fad7210bb2de21e602e55dcdb4568acbfcd82daf928b0458c8407a8bf4ad9c6R24
Does it answer your question?

Reply:
Sorry I was not clear enough. I mean instead of having `Tree` implement `HasComments` as you did, why didn't you keep it has before and in your rule do a similar check as above `if (tree instanceof HasComments)` to run the check on only on Tree element with comments ?

Reply:
Ok, now I get the point. Well, first when I implemented the check I was thinking about how to subscribe on `HasComments` and the natural way for me was to change the class hierarchy. This class I needed to update later and here was another possible solution. Let's discuss offline that

Reply:
Ok, let's apply a simpler design here.